### PR TITLE
Added some py312 subports

### PR DESCRIPTION
**Pull request by catap**
#### Description

Depends on https://github.com/macports/macports-ports/pull/21733

###### Type(s)

- [x] enhancement

###### Tested on

macOS 12.7.1 21G920 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Review by lpsinger — APPROVED** [review]

File: python/py-flake8/Portfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ checksums           rmd160  3d26d25023445ef6462dabb50503736d19c3c597 \
                     sha256  c61007e76655af75e6785a931f452915b371dc48f56efd765247c8fe68f2b181 \
                     size    138524
 
-python.versions     37 38 39 310 311
+python.versions     37 38 39 310 311 312
```
Comment (reneeotten) [inline python/py-flake8/Portfile:31]: you have to add the select file as well ; that is the case for some of the other ports as well. Please verify.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-flake8/Portfile (hunk 1 above)
Comment (catap) [reply python/py-flake8/Portfile:31]: @reneeotten can we start from https://github.com/macports/macports-ports/pull/21733?

the only reason why I’ve spilt it into servers PR (and a few is in the queue): GitHub has its limit 30 or a bit less commits

Comment (catap) [reply python/py-flake8/Portfile:31]: BTW I've tacked your comment into account and this is fixed and all affected ports are verified.

Comment (reneeotten) [reply python/py-flake8/Portfile:31]: I merged the initial PR - please rebase this one.

Comment (catap) [reply python/py-flake8/Portfile:31]: Here it is. Rebase also included verify that files for select group are added.

**Comment by catap** [comment]
@reneeotten just a reminder about this PR

**Comment by catap** [comment]
Thanks, Perry!

**Pull request merged**